src/renderer/components/Terminal/components/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,193 @@
+import { Typography } from '@mui/material';
+import { ipcRenderer } from 'electron';
+import React, { useState } from 'react';
+import { CHANNELS } from 'src/shared/types/api';
+import { LocalPublication } from 'src/shared/types';
+import 'xterm/css/xterm.css';
+import {
+  Block,
+  DeleteSweep,
+  PlayArrow,
+  RestartAlt,
+  RotateLeft,
+} from '@mui/icons-material';
+import Tooltip from 'src/renderer/components/Tooltip/Tooltip';
+import IconButton from 'src/renderer/components/IconButton/IconButton';
+import { useTranslation, CustomTypeOptions } from 'react-i18next';
+import * as Styled from './style';
+
+type Props = {
+  project: LocalPublication;
+};
+
+enum SERVER {
+  IDLE = 'idle',
+  INSTALLING = 'installing',
+  SERVER_RUNNING = 'running',
+  CLEARING_CACHE = 'clearing',
+  STOPPING = 'stopping',
+}
+
+const Toolbar: React.FC<Props> = ({ project }) => {
+  const [state, setState] = useState<SERVER>(SERVER.IDLE);
+  const { t } = useTranslation('translation');
+
+  const canStopProcess = state === SERVER.IDLE || state === SERVER.STOPPING;
+  const canStartServer = state !== SERVER.IDLE;
+  const canRestartServer = state !== SERVER.SERVER_RUNNING;
+  const canHardResetServer = state !== SERVER.SERVER_RUNNING;
+  const canClearCache = state !== SERVER.IDLE;
```

Comment:
This logic was very confusing to me until I scrolled down to see how it was used :haha:

I'd change the naming pattern to something like `isXDisabled`. Or just reverse the logic and negate it when passing down to the components - whatever you think is more intuitive.